pkg/ipam/metadata/manager.go

```diff
@@ -116,8 +118,25 @@ func (m *Manager) GetIPPoolForPod(owner string) (pool string, err error) {
 		return "", fmt.Errorf("failed to lookup pod %q: %w", namespace+"/"+name, err)
 	} else if !ok {
 		return "", &ResourceNotFound{Resource: "Pod", Namespace: namespace, Name: name}
-	} else if ipPool, hasAnnotation := pod.Annotations[annotation.IPAMPoolKey]; hasAnnotation {
-		return ipPool, nil
+	} else {
+		switch family {
```

Comment:
nit: I think it might be worth splitting this out into a function that takes the familiy and the annotations as arguments, and returns `(pool string, ok bool)`

Reply:
OK, I will split them out into a function. Thanks!